### ISSUE TITLE
SKOS Visualization

### ISSUE THREAD

**Issue by giraygi** [EBISPOT/OLS#503]
Currently, only the schema of SKOS ontologies can be displayed with OLS. As a result, the list of concepts in a SKOS ontology are displayed as instances of the term "concept". However, concepts of a SKOS ontology should be browsed in a similar fashion with terms of an OWL ontology in visualization. I wonder if this will be possible at some point.
Cheers,
Giray

**Comment by henrietteharmse** [comment]
@giraygi Certainly. We are keen to look into this. Do you have a minimal complete SKOS ontology illustrating the problem you experiencing?

**Comment by giraygi** [comment]
@henrietteharmse Thank you very much for your reply! [UAT Ontology](https://terminology.nfdi4ing.de/ts4ing/ontologies/uat) can be a good example to this. The concept instances are displayed with numbers here (most like other skos ontologies that we tried earlier) and the prefLabel field of each "concept instance" gives the actual concept name that we would prefer to see in the display instead of the number. We had tried to set the config variable "skos" to "yes" but we did not see any change in display (we removed it afterwards).

**Comment by henrietteharmse** [comment]
@giraygi That is actually a very useful example. 

I think the first thing that strikes me is that I think it should be possible currently to cater for using prefLabel as label for your concept instances. We can investigate and hopefully correct this.

The second bigger challenge is to cater for a concept hierarchy. To make this concrete I will refer to an example from UAT, 1040. For easy reference I added a screenshot below.

![image](https://user-images.githubusercontent.com/32327979/135051546-0248b949-00d8-4c99-88a8-ea9241ed404e.png)

Based on this example, I suspect a partial concept hierarchy will need to look as follows:
`
1041
	1040
		1528
1048
	1040
		1528
` 

Do you agree with representation?

This is not something that can be easily implemented in OLS currently. However, we do hope to do a redesign of OLS in the near future and will definitely keep this in mind.

**Comment by henrietteharmse** [comment]
@giraygi Glad you are able to sort the label issue!
Thanks fixing my hierarchy example - I seem to mix up the interpretation of broader and narrower routinely.
I have added this issue to our milestone 4.x for consideration for inclusion.

**Comment by giraygi** [comment]
@henrietteharmse 

Many thanks for your interest! We are looking forward to its solution in milestone 4.x!

**Comment by cmungall** [comment]
Another approach here would be to have a standard configurable skos2owl procedure that groups could execute as part of their release pipelines upstream of OLS. They would locally configure e.g. which predicates (if any) are translated to existentials.

One advantage of this is uniformity across different browsers/APIs rather than each one implementing their own slightly varying skos2owl

**Comment by nichtich** [comment]
@cmungall We basically follow the opposite approach by mapping OWL and other ontologies (GND ontology, Wikidata ontology...) to [(a slight superset of) SKOS](https://gbv.github.io/jskos/) to unify basic properties of [knowledge organization systems](https://www.isko.org/cyclo/kos) such as "concepts", "hierarchy", "preferred labels...". This is what SKOS was created for.
